latex.bzl

```diff
@@ -10,59 +12,140 @@ LatexOutputInfo = provider(
+
+    Latex and a set of scripts and binaries in the tool suite
+    makes use of a library, kpathsea.
+    In general latex distributions are encouraged to follow the 'TDS'
+    structure. And tools and script might make assumptions that the
+    layout of directories respects that structure.
+
+    But from our perspective, trying to shoehorn the partioned
+    repositories according
+    (E.g. <sandbox>/bazel-latex/external/<ext_texlive_repo>/<files>)
+    to the TDS creates an unnecessary complexity.
+    Also kpathsea tries to be efficient about looking up files.
+    So to derive if a folder is of interest, kpathsea checks if the number
+    of files or folders in the current folder being inspected is greater
+    than 2.
+    Unfortunately symlinks are (currently) not counted. And Bazel makes heavy
+    use of symlinks.
+
+    However, kpathsea makes heavy use of environment variables (and
+    ls-R database, IIRC).
+    So we can work around this limitation. Also, by only adding the
+    search paths of the files mapped to the environment variable we can reduce
+    the search space, and reduce build times.
+    https://tug.org/texinfohtml/kpathsea.html#Supported-file-formats,
+    lists all environment variables one can set.
+
+    Args:
+      ctx: For accessing the inputs parameters.
+      dep_tools: the tools needed by latexrun
+      files: all files that might be needed as part of the build.
+
+    Returns:
+      A list of commands to provide to the ctx.actions.run invokation
+    """
+
+    def filter_on(files, exts):
+        directories = []
+        for inp in files:
+            dirname = inp.dirname
+            valid = False
+            if not exts:
+                valid = True
+            else:
+                for ext in exts:
+                    if inp.path.endswith(ext):
+                        valid = True
+            if valid and dirname not in directories:
+                directories.append(dirname)
+        return directories
+
+    type_env = {
+        "AFMFONTS": ".:" + ":".join(filter_on(files, [".afm"])),
+        "BIBINPUTS": ".:" + ":".join(filter_on(files, [".bib"])),
+        "ENCFONTS": ".:" + ":".join(filter_on(files, [".enc"])),
+        "LUAINPUTS": ".:" + ":".join(filter_on(files, [".lua" or ".luc"])) +
+                     ":$TEXINPUTS:",
+        "OPENTYPEFONTS": ".:" + ":".join(filter_on(files, [])),
+        "T1FONTS": ".:" + ":".join(filter_on(files, [".pfa", ".pfb"])),
+        "TEXFONTMAPS": ".:" + ":".join(filter_on(files, [".map"])) + ":",
+        "TEXINPUTS": ".:" + ":".join(filter_on(files, [])),
+        "TFMFONTS": ".:" + ":".join(filter_on(files, [".tfm"])),
+        "TTFONTS": ".:" + ":".join(filter_on(files, [".ttf", ".ttc"])),
+        "VFFONTS": ".:" + ":".join(filter_on(files, [".vf"])),
+    }
+    env = {
+        "PATH": ":".join(
+            [
+                dep_tools[0].to_list()[0].dirname,  # latex bin folder
```

Comment:
dep_tools doesn't make much sense anymore when the tool wrapper is gone, consider dropping it and use the toolchain directly

Reply:
done